bin/cmsl1t

```diff
@@ -0,0 +1,128 @@
+
+    input_files = config.get('input', 'files')
+    click.echo("Input files:")
+
+    if len(input_files) > 10:
+        click.echo(input_files[:10], "... and",
+                   len(input_files) - 10, "more files")
+    else:
+        click.echo(input_files)
+
+    from cmsl1t.playground.eventreader import EventReader
+    reader = EventReader(input_files, events=nevents)
+
+    results = [analyzer.prepare_for_events(reader) for analyzer in analyzers]
+    check(results, analyzers, 'prepare_for_events')
+
+    click.echo(section.format("Processing events"))
+    # Fill the histograms from the tuples
+    counter_rate = nevents / 10 if nevents <= 10000 else 1000
+    for entry, event in enumerate(reader):
+        if entry % counter_rate == 0:
+            click.echo("{} of {}".format(entry, nevents))
+        results = [analyzer.process_event(entry, event)
+                   for analyzer in analyzers]
+        check(results, analyzers, 'process_event')
+        if all(results) is not True:
+            break
+
+
+def run(config, nevents, reload_histograms):
+    results = [False]
+    # Fetch the analyzer
+    analyzers = config.get('analysis', 'analyzers')
+    analyzers = [load_analyzer(analyzer)(config) for analyzer in analyzers]
+    click.echo(analyzers)
+
+    if not reload_histograms:
+        process_tuples(config, nevents, analyzers)
+    else:
+        # TODO: Support multiple input root histogram files
+        pass
+        # isOK = analyzer.reload_histograms(args.inputs[0])
+        # if isOK is not True:
+        #     return "Problem in reload_histograms() with analyzer: " \
+        #         + str(args.analysis)
+    # Write out the histograms
+    for analyzer in analyzers:
+        analyzer.write_histograms()
+
+    # Turn the histograms to plots
+    click.echo(section.format("Making plots"))
+    results = [analyzer.make_plots() for analyzer in analyzers]
+    check(results, analyzers, 'make_plots')
+
+    # Finalize
+    print(section.format("Finalizing things"))
+    results = [analyzer.finalize() for analyzer in analyzers]
+    check(results, analyzers, 'finalize')
+
+    return all(results)
+
+
+def check(results, analyzers, method):
+    msg = 'Problem during {method}() with analyzer "{analyzer}"'
+    if all(results) is not True:
+        for i, r in enumerate(results):
+            if r is not True:
+                click.echo(msg.format(method=method, analyzer=analyzers[i]))
+
+
+@click.command()
+@click.option('-c', '--config_file', help='YAML style config file', type=click.File())
+@click.option('-n', '--nevents', default=-1, help='Number of events to process.')
+@click.option('-r', '--reload-histograms', is_flag=True,
+              help="Reload histograms from a file and skip the input tuples")
+def analyze(config_file, nevents, reload_histograms):
+    click.echo(section.format("Starting CMS L1T Analsysis"))
+    click.echo([config_file, nevents, reload_histograms])
```

Comment:
Is this supposed to be meaningful or is it left over debugging?  If it's meant to mean something, could we say what the variables mean:
```py
click.echo(["config file: ",config_file])
click.echo(["No. events:",nevents])
# etc
```